Ya se muestran las materias con codigo alfanumerico

juli-rivero opened the pull request:
El problema, ademas del dicho en el [issue](#18), fue que los topics estan en minuscula mientras que los códigos de las materias están en mayusculas.

Cambie por completo ese pequeño trozo de código en `MainApp.js` - `21    const materias = React.useMemo...` con el uso de [new Map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map). 

Creo que quedó con código más reducido, más legible y espero que con mejor performance. Y, por más que el código en esa parte cambió bastante, el funcionamiento es el mismo con la excepción del issue arreglado ;D

> [!NOTE]
> Tambien tuve que cambiar detalles de otros archivos (ver archivos cambiados) por lo de mayusculas y minusculas y porque los repositorios estaban con Set en vez de Arrays

Resolves #18

juli-rivero committed:
Update materias.json con materias 2024 Ing. Informática

Añadidos las materias que me aparecen a mi (mi carrera es Ingeniería en Informática) en https://guaraniautogestion.fi.uba.ar/g3w/oferta_comisiones.
Añadir las de cada uno es facil, solamente hay que correr este script que hice en `inspeccionar -> consola`:
    const periodos = document.querySelectorAll("[periodo]");
    
    const data = new Map()
    
    periodos.forEach(periodo => {
      const materias = periodo.querySelectorAll("[actividad]")
      materias.forEach(materia => {
        const indiceParentesis = materia.getAttribute("actividad").indexOf("(");
        const materiaId = materia.getAttribute("actividad").slice(indiceParentesis + 1, -1);
        const materiaName = materia.getAttribute("actividad").slice(0, indiceParentesis).trim();
        data.set(materiaId, materiaName);
      })
    })
    console.log(Object.fromEntries(data))

Luego, solo hay que hacer click derecho -> "Copiar Objeto"

juli-rivero committed:
Aparecen los nombres de todas las materias (incluidas las de los codigos alfanumericos) en la lista de materias

juli-rivero committed:
Aparecen los nombres de todas las materias (incluidas las de los codigos alfanumericos) en la lista de materias

juli-rivero committed:
cambiado reponames de tipo de dato Set a Array

juli-rivero committed:
cambiado reponames de tipo de dato Set a Array

juli-rivero committed:
Ahora se muestran los repositorios de las materias con codigo alfanumerico

juli-rivero committed:
Ahora se muestran los repositorios de las materias con codigo alfanumerico

juli-rivero committed:
formateo de codigo trabajado con prettier

juli-rivero committed:
formateo de codigo trabajado con prettier

juli-rivero committed:
correccion aparecían los repositorios de materias nuevas al estar el filtro fiubaOnly

juli-rivero committed:
correccion aparecían los repositorios de materias nuevas al estar el filtro fiubaOnly

juli-rivero committed:
Merge branch 'master' of github.com:juli-rivero/FIUBA-Repos

FdelMazo commented:
Gracias Juli! El finde lo reviso

FdelMazo reviewed (approved):
Hola @juli-rivero , el PR lo veo muy bien, gracias por el cambio!

Mis dos comentarios son:
- Como lo testeaste? Hay alguna materia que antes no se veia, que ahora si se vea?
- Me acabo de dar cuenta que en las materias introducidas en #17, quedaron todas en mayusculas cuando el resto de las materias estan tituladas correctamente. Perdón que lo veo recien ahora, pero podrías modificarlo para mantenerlo consistente?

juli-rivero commented:
Buenas fede! 

Respecto a los nombres de las materias, yo tampoco me había dado cuenta, en el SIU vienen en mayúsculas y no me dí cuenta de cambiarlas. Mañana los "capitalizo" ;D

Los códigos de las materias las dejo como estan (en mayusculas) o lo cambio todo a minusculas? Lo digo porque me parece lo correcto que esté en mayusculas (y ya adapté el código a eso), pero como los topics en github se ponen todos en minusculas sería un poquito más optimo tenerlos en minusculas

Lo testeé con "Introducción al desarrollo de software" y con "Fundamentos de programacion", que son las dos materias que tengo repos propios (No me dí cuenta de poner eso en el commit anterior, perdon por eso)

FdelMazo commented:
> Los códigos de las materias las dejo como estan (en mayusculas) o lo cambio todo a minusculas? Lo digo porque me parece lo correcto que esté en mayusculas (y ya adapté el código a eso), pero como los topics en github se ponen todos en minusculas sería un poquito más optimo tenerlos en minusculas

Esto me parece bien así como lo dejaste! Los códigos en mayusculas, y que el proyecto se adapte a eso con las conversiones

juli-rivero committed:
Añadida carpeta guia para subir materias

1. Se "capitalizaron" los palabras de los nombres de las materias (primera letra de cada palabra en mayúscula)
2. Cree una carpeta que tiene un README.md con los pasos a seguir para obtener las materias y actualizar el json (decime si esta bien o si está de más y lo saco)

FdelMazo commented on src/data/obtener-subir-materias/README.md:
Muy buena idea tener este archivo! Lo único que te pido es si en vez de ponerlo dentro de `src/data` lo pones en otro lado (`/scripts`? donde digas). Esto es porque todo lo que esta en `src` se compila y se usa en el sitio, y mientras mas archivos tiene, más pesado va a ser el sitio web. Como esto es algo anexo (un script de python), tal vez queda mejor en otro lado del repo, fuera de `src`.

juli-rivero replied:
Tenes razón, mañana por la tarde lo muevo y también tengo que cambiar los nombres para que las palabras: de, la, las, el, los, para, etc. no aparezcan con mayúscula inicial

juli-rivero replied:
Creo que ahí esta, revisalo con tiempo y decime si cambio o agrego algo, ya que hasta el 19 (que empiezan las clases) tengo bastante tiempo libre ;D

juli-rivero committed:
formateo del json por separado

juli-rivero committed:
obtener-subir-materias afuera del src

juli-rivero committed:
actualizado README del obtener-subir-materias

juli-rivero commented:
Una funcionalidad que cambia con lo hecho es que ahora solo se muestran los códigos de las materias que tienen algun topic en algun repo

Me explico:
Supongamos que hay una materia "Materia01" que tiene los códigos ["000", "001", "002"], pero solo aparecen dos repositorios con los codigos "000" y "002"; entonces, en la página se mostraría:

> `000`  `002`  -  2 Repos
> Materia01

Es algo que no me había dado cuenta que cambie y que debería volverlo a como estaba

FdelMazo commented:
Quedo muy bueno @juli-rivero , muchas gracias
